src/createHeadersInterceptor.ts

```diff
@@ -0,0 +1,10 @@
+import type { Interceptor } from "@bufbuild/connect";
+
+export function createHeadersInterceptor(headers: any): Interceptor {
```

Comment:
Looks like if we implement `createHeaderInterceptor`

We could refactor `createAuthInterceptor` to use this method instead.

https://github.com/substreams-js/substreams-js/blob/main/packages/core/src/auth/create-auth-interceptor.ts